extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/webjar/WebJarUtil.java

```diff
@@ -0,0 +1,273 @@
+package io.quarkus.vertx.http.deployment.webjar;
+
+import java.io.ByteArrayInputStream;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.FileSystem;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.builder.Version;
+import io.quarkus.deployment.pkg.builditem.CurateOutcomeBuildItem;
+import io.quarkus.fs.util.ZipUtils;
+import io.quarkus.maven.dependency.ArtifactKey;
+import io.quarkus.maven.dependency.ResolvedDependency;
+import io.quarkus.paths.PathCollection;
+import io.quarkus.runtime.ApplicationConfig;
+
+/**
+ * Utility for Web resource related operations
+ */
+public class WebJarUtil {
+
+    private static final Logger LOG = Logger.getLogger(WebJarUtil.class);
+
+    private static final String CUSTOM_MEDIA_FOLDER = "META-INF/branding/";
+    private static final List<String> OVERRIDABLE_RESOURCES = Arrays.asList("logo.png", "favicon.ico", "style.css");
+    private static final String CSS = ".css";
+
+    private WebJarUtil() {
+    }
+
+    static Path copyResourcesForDevOrTest(CurateOutcomeBuildItem curateOutcomeBuildItem, ApplicationConfig config,
+            WebJarBuildItem webJar,
+            ResolvedDependency resourcesArtifact,
+            Path deploymentBasePath)
+            throws IOException {
+
+        Path deploymentPath = Files.createDirectories(deploymentBasePath);
+
+        PathTargetVisitor visitor = new PathTargetVisitor(deploymentPath);
+        copyResources(curateOutcomeBuildItem, config, webJar, resourcesArtifact, visitor);
+
+        return deploymentPath;
+    }
+
+    static Map<String, byte[]> copyResourcesForProduction(CurateOutcomeBuildItem curateOutcomeBuildItem,
+            ApplicationConfig config, WebJarBuildItem webJar,
+            ResolvedDependency resourcesArtifact)
+            throws IOException {
+
+        InMemoryTargetVisitor visitor = new InMemoryTargetVisitor();
+        copyResources(curateOutcomeBuildItem, config, webJar, resourcesArtifact, visitor);
+
+        return visitor.getContent();
+    }
+
+    private static void copyResources(CurateOutcomeBuildItem curateOutcomeBuildItem, ApplicationConfig config,
+            WebJarBuildItem webJar,
+            ResolvedDependency resourcesArtifact, WebJarResourcesTargetVisitor visitor)
+            throws IOException {
+        final ResolvedDependency userApplication = curateOutcomeBuildItem.getApplicationModel().getAppArtifact();
+
+        ClassLoader classLoader = WebJarUtil.class.getClassLoader();
+
+        for (Path p : resourcesArtifact.getResolvedPaths()) {
```

Comment:
We should be moving away from using resolved path to explore the content of a resolved dependency switching to `PathTree userApplication.getContentTree()`. Then you walk the tree or process a specific path.

Reply:
done